File: fileshare/src/main/java/alkfejl/bead/fileshare/model/File.java
package alkfejl.bead.fileshare.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "FILE")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class File extends BaseEntity {
    @Column(nullable = false, unique = true)
    private String fullPath;

    @Column(nullable = false)
    private String path;

    @Column
    private String fileName;

    @Column(nullable = false)
    private int editLevel;

    @Column(nullable = false)
    private int viewLevel;

    @JoinColumn(nullable = false)
    @ManyToOne(targetEntity = User.class)
    private User owner;

    @Column
    private boolean isDir;

    @Override
    public String toString(){
        if(!(this.isDir)) {
            return this.fileName;
        } else {
            return this.fileName;
        }
    }

}

File: Angular/fileshare/src/app/classes/file.ts
import {User} from "./user";

export class File {

  private _id: number;
  private _fullPath: string;
  private _path: string;
  private _fileName: string;
  private _editLevel: number;
  private _viewLevel: number;
  private _owner: User;
  private _isDir: boolean;

  constructor(id: number, fullPath: string, path: string, fileName: string, editLevel: number, viewLevel: number, owner: User, isDir: boolean) {
    this._id = id;
    this._fullPath = fullPath;
    this._path = path;
    this._fileName = fileName;
    this._editLevel = editLevel;
    this._viewLevel = viewLevel;
    this._owner = owner;
    this._isDir = isDir;
  }


  public get id(): number {
    return this._id;
  }

  public get fullPath(): string {
    return this._fullPath;
  }

  public get path(): string {
    return this._path;
  }

  public get fileName(): string {
    return this._fileName;
  }

  public get editLevel(): number {
    return this._editLevel;
  }

  public get viewLevel(): number {
    return this._viewLevel;
  }

  public get owner(): User {
    return this._owner;
  }

  public get isDir(): boolean {
    return this._isDir;
  }
}
